alerts/cluster-monitoring-operator/TargetDown.md

```diff
@@ -0,0 +1,269 @@
+metric = 0) for at least 15 minutes.
+
+## Impact
+
+Metrics from the offline targets are not scraped by Prometheus. By consequence,
+we lose the visibility into the health and performance of the monitored application
+represented by the target.
+
+The alerts may lose accuracy or even become crippled if the missing metrics play
+a role in the triggering condition of the alert. Critical alerts may not fire
+when they should, potentially leading to unnoticed service disruption.
+
+Auto-scalers may not work properly if the missing metrics are critical to scaling
+conditions. This leads to potential degradation in user experience or waste of
+valuable resources.
+
+## Diagnosis & Mitigation
+
+The `TargetDown` alert indicates that one or more scrape targets are not responding
+to Prometheus scrape query.
+
+The alert has labels `namespace`, `service` and `job`. With these labels we can
+identify which Prometheus instance fails to scrape which target.
+
+In the Openshift web console, go to the tab **Observe** -> **Metrics**.  Check
+the metric `up` with these labels to find the down targets are in which pods.
+Here is an example query:
+```text
+up{namespace="openshift-monitoring", service="thanos-querier", job="thanos-querier"} == 0
+```
+
+Then check the label `prometheus` and `pod` in the returned metric `up`.
+The label `prometheus` should be either `openshift-monitoring/k8s` or `openshift-user-workload-monitoring/user-workload`,
+indicating the Prometheus pods scraping the target are either `prometheus-k8s-0`
+/ `prometheus-k8s-0` in the namespace `openshift-monitoring`, or `prometheus-user-workload-0`
+/ `prometheus-user-workload-1` in the namespace `openshift-user-workload-monitoring`.
+The label `pod` indicates the pod exposing the metrics endpoint.
+
+The targets are defined in either `ServiceMonitor` or `PodMonitor` resources.
+The labels `namespace` and `service` that help us locate the `Service` exposing
+the metric endpoint.
+The label `namespace` and `job` can locate the `ServiceMonitor` or `PodMonitor` that
+configures Prometheus to scrape the target. The `job` label is the name of the monitor.
+
+Now we have both end of the metric scraping flow as well as the monitor resources
+linking them together.
+We are ready to diagnosing the root cause by inspecting each component on the scraping
+workflow.
+
+
+### Network Issues
+
+Check the network connectivity between the Prometheus pod and the target pod.
+Ensure that the target pod is reachable from the Prometheus pod and that there are
+no firewall rules or network interruptions blocking communication.
+
+Please refer to [Troubleshooting network issues](https://docs.openshift.com/container-platform/4.13/support/troubleshooting/troubleshooting-network-issues.html)
+in Openshift document for more details.
+
+### Target Resource Exhaustion
+
+Check if the target pod resource utilization is too high, causing it to become unresponsive.
```

Comment:
I'd start by asking to make sure that the metrics are available locally first:

after checking the pod/container is healthy and running (check its logs, events, ) we can:

on the container that exposes the metric, curl its localhost:port/metrics or whatever couple of times and see if the metrics are returned.

Reply:
if metrics are not return then it's an application problem that needs to be debugged (add/enable debug logs, check existing logs, check if any `Rate Limiting or Throttling` (we can merge with that section here) ).

Reply:
will merge the section Rate Limiting or Throttling here